tests/unit/cli/reporter.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { createReport } from '../../../src/cli/reporter';
+import { Reporter } from '../../../src/types';
+import { writeFile } from '../../../src/utils';
+import { table } from 'table';
+import { blue, green, red, yellow } from 'kleur/colors';
+
+const mockCompatData = {
+  'test-package-1': {
+    compatible: true,
+    range: '>=8.0.0',
+  },
+  'test-package-2': {
+    compatible: false,
+    range: '^6.0.0',
+  },
+  'test-package-3': {
+    compatible: undefined,
+    range: 'n/a',
+  },
+};
+
+jest.mock('../../../src/utils', () => ({
+  writeFile: jest.fn(),
+}));
+
+jest.mock('table', () => ({
+  table: jest.fn(),
+}));
+
+const htmlExpected =
+  '<!DOCTYPEhtml><htmllang="en"><head><style>h1{font-family:arial,sans-serif;}table{font-family:arial,sans-serif;border-collapse:collapse;width:100%;}td,th{border:1pxsolid#dddddd;text-align:left;padding:8px;}tr:nth-child(even){background-color:#dddddd;}.red{color:#ff0000;}.green{color:#0f9b4e;}.yellow{color:#ce8d02;}</style><title>depngn</title></head><body><h1>Nodeversion:8.0.0</h1><table><tr><th>package</th><th>compatible</th><th>range</th></tr><tr><td>test-package-1</td><tdclass="green">true</td><td>>=8.0.0</td></tr><tr><td>test-package-2</td><tdclass="red">false</td><td>^6.0.0</td></tr><tr><td>test-package-3</td><tdclass="yellow">undefined</td><td>n/a</td></tr></table></body></html>';
+
+describe('createReport', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('outputs correct table', async () => {
+    const originalConsoleLog = console.log;
```

Comment:
May we move this logic to the `beforeEach` and `afterEach` functions?

Reply:
oh yeah good call!

usually, i like to keep these in each individual test -- to keep things explicit and obvious -- but for these tests, we'll always have to mock `console.log`, so i think it makes sense.